File: SalyanthanSchool.Core/Entities/Student.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

[Table("Student")]
public class Student
{
    [Column("id")]
    public int Id { get; set; }

    //[Required]
    //[Column("admission_no")]
    //public string AdmissionNo { get; set; } = null!;

    [Column("first_name")]
    public string FirstName { get; set; } = null!;

    [Column("middle_name")]
    public string? MiddleName { get; set; }   

    [Column("last_name")]
    public string LastName { get; set; } = null!;

    [Column("gender")]
    public string Gender { get; set; } = null!;

    [Column("blood_group")]
    public string? BloodGroup { get; set; }

    [Column("date_of_birth")]
    public DateTime? DateOfBirth { get; set; }

    [Column("admission_date")]
    public DateTime AdmissionDate { get; set; }   

    [Column("address")]
    public string? Address { get; set; }

    [Column("guardian_name")]
    public string? GuardianName { get; set; }

    [Column("guardian_contact")]
    public string? GuardianContact { get; set; }

    [Column("grade_id")]
    public int GradeId { get; set; }

    [Column("section_id")]
    public int SectionId { get; set; }

    [Column("photo")]
    public string? Photo { get; set; }

    [Column("is_active")]
    public bool IsActive { get; set; } = true;

    [Column("created_at")]
    public DateTime? CreatedAt { get; set; } = DateTime.Now;

    [Column("updated_at")]
    public DateTime? UpdatedAt { get; set; }
}

File: client/app/dashboard/types/student.ts
export type Student = {
  id: number;

  firstName: string;
  middleName?: string | null;
  lastName: string;

  fullName: string;

  rollNo?: number;            

  gender?: "Male" | "Female" | "Other" | null;
  bloodGroup?: "A+" | "A-" | "B+" | "B-" | "AB+" | "AB-" | "O+" | "O-";

  dateOfBirth?: string | null;
  admissionDate: string;

  address?: string | null;
  guardianName?: string | null;
  guardianContact?: string | null;

  gradeId: number;
  gradeName?: string | null;

  
  sectionId: number;
  sectionName?: string | null;
  
  photo?: string | null;

  isActive: boolean;
  createdAt: string;
};

// CREATE (INSERT)
export type CreateStudentPayload = {
  firstName: string;
  middleName?: string;
  lastName: string;

  gender: string;
  bloodGroup?: string;

  dateOfBirth?: string;
  admissionDate: string;

  address?: string;
  guardianName?: string;
  guardianContact?: string;

  gradeId: number;
  sectionId: number;

  isActive: boolean;

  photoFile?: File;
};

// UPDATE (PUT)
export type UpdateStudentPayload = {
  id: number;

  firstName: string;
  middleName?: string;
  lastName: string;

  gender: string;
  bloodGroup?: string;

  dateOfBirth?: string;
  admissionDate: string;

  address?: string;
  guardianName?: string;
  guardianContact?: string;

  gradeId: number;
  sectionId: number;

  isActive: boolean;

  photoFile?: File;
};

// PATCH (OPTIONAL)
export type PatchStudentPayload = {
  id: number;
  firstName?: string;
  middleName?: string;
  lastName?: string;
  gender?: string;
  bloodGroup?: string;
  dateOfBirth?: string;
  address?: string;
  guardianName?: string;
  guardianContact?: string;
  gradeId?: number;
  sectionId?: number;
  isActive?: boolean;
  photoFile?: File;
};
// QUERY PARAMETERS
export type StudentQueryParameters = {
  pageNumber?: number;
  pageSize?: number;
  search?: string;
  sortBy?: string;
  sortDir?: "asc" | "desc";
  isActive?: boolean;
  gradeId?: number;
  sectionId?: number;
};

// PAGINATED RESPONSE
export type StudentPaginatedResponse = {
  data: Student[];
  meta: {
    pageNumber: number;
    pageSize: number;
    total: number;
  };
};
